src/main/java/com/yahoo/bullet/rest/controller/HTTPQueryController.java

```diff
@@ -23,18 +31,33 @@
     @Autowired @Setter
     private QueryService queryService;
 
+    private static final FieldTypeAdapterFactory<Clause> CLAUSE_FACTORY =
+            FieldTypeAdapterFactory.of(Clause.class, t -> t.getAsJsonObject().get(Clause.OPERATION_FIELD).getAsString())
+                    .registerSubType(FilterClause.class, Clause.Operation.RELATIONALS)
+                    .registerSubType(LogicalClause.class, Clause.Operation.LOGICALS);
+    private static final Gson GSON = new GsonBuilder().registerTypeAdapterFactory(CLAUSE_FACTORY)
+            .excludeFieldsWithoutExposeAnnotation()
+            .create();
+
     /**
      * The method that handles POSTs to this endpoint. Consumes the HTTP request, invokes {@link QueryService} to
      * register and transmit the query to Bullet.
      *
-     * @param query The JSON query.
+     * @param queryString The JSON query.
      * @return A {@link CompletableFuture} representing the eventual result.
      */
     @PostMapping(path = "${bullet.endpoint.http}", consumes = { MediaType.TEXT_PLAIN_VALUE }, produces = { MediaType.APPLICATION_JSON_VALUE })
-    public CompletableFuture<String> submitHTTPQuery(@RequestBody String query) {
+    public CompletableFuture<String> submitHTTPQuery(@RequestBody String queryString) {
         HTTPQueryHandler queryHandler = new HTTPQueryHandler();
         String queryID = QueryService.getNewQueryID();
-        queryService.submit(queryID, query, queryHandler);
+        // Remove window information from queryString since we don't support windowing for this endpoint.
+        try {
+            Query query = GSON.fromJson(queryString, Query.class);
```

Comment:
Let's not do the full parsing. Just do a non-Query aware generic JSON parsing into a Map<String, Object> and delete from Map the "window" key and make JSON again (using JSONFormatter).

Reply:
Done.